boat_simulator/common/unit_conversions.py

```diff
@@ -1,10 +1,15 @@
 """Unit conversion logic, mostly contained in the `UnitConverter` class."""
 
 from __future__ import annotations
-from boat_simulator.common.types import EnumAttr, Scalar
+
+import math
 from enum import Enum
 from typing import Dict
 
+import numpy.typing as npt
```

Comment:
Since we only seem to be using `ArrayLike` from this package, it's probably cleaner to just import ArrayLike and change all the `npt.ArrayLike` lines to `ArrayLike`

Reply:
The type specified to `NDArray` must be a `numpy` recognized type. I don't think we can plug and play our own defined `Scalar` here. I tried myself and look like it's not happy. A list of accepted datatypes is here: https://numpy.org/doc/stable/reference/arrays.scalars.html#built-in-scalar-types.

You can specify `np.object_` as a datatype into `NDArray` (`NDArray[np.object_]` and then feed in `Scalars`. But, this seems like an over-engineered solution when `ArrayLike` exists.

There is a way to set `numpy` array datatypes to support ints or floats only, but it doesn't work with our `Scalar` object. The way to do it is outlined in the answer to [this StackOverflow post ](https://stackoverflow.com/questions/70714087/how-to-typehint-numpy-array-of-floats-and-or-integers), but it introduces more code. For these reasons, I'm just going to stick with `ArrayLike`.

Reply:
You can do `Union[Scalar, NDArray]` if you are worried about typing

Reply:
You could also create a new custom type inside the boat simulator's typing module and import it from there.

Reply:
The solution we came to is to create a new `ScalarOrArray` type for the `Union[Scalar, NDArray]` type.